### PR TITLE
Add an E2E test for the spot termination handler

### PR DESCRIPTION
**Pull request by aermakov-zalando**
This is now fairly easy to test, and we'll know when it's broken again (see #4167 for example).

### CODE REVIEW

**Comment by aermakov-zalando** [comment]
👍

File: test/e2e/cluster_config.sh [hunk 1]
```diff
@@ -85,6 +85,15 @@ clusters:
     profile: worker-default
     min_size: 0
     max_size: 21
+  - discount_strategy: spot
+    instance_types: ["m4.large", "m5.large", "m5.xlarge", "m4.xlarge"]
```
Comment (jonathanbeber) [inline test/e2e/cluster_config.sh:89]: does it respect order? If so why we go with m4 first instead of m5?

Comment (aermakov-zalando) [reply test/e2e/cluster_config.sh:89]: IIRC it's whatever has the least potential for disruption, I'm not sure if order matters. It also doesn't matter anyway, the node would run a pause container and a small shell script and then get cordoned.

Comment (jonathanbeber) [reply test/e2e/cluster_config.sh:89]: Got it. Ofc it does not matter for the process, just thinking that `m5`s are cheaper.

**Comment by jonathanbeber** [comment]
:+1:

File: test/e2e/cluster_config.sh (hunk 1 above)
Comment (jonathanbeber) [reply test/e2e/cluster_config.sh:89]: but we can't be sure for spots anyway, just ignore.

**Pull request merged**